addons/block_code/ui/blocks/block/block.gd

```diff
@@ -88,6 +88,12 @@ func get_parameter_values() -> Dictionary:
 	return template_editor.get_parameter_values()
 
 
+## Use the current BlockEditorContext components
+func refresh_context():
+	if _context and _block_extension:
+		_block_extension.context_node = _context.parent_node
```

Comment:
Can this be a signal handler for `_context.changed`, instead, or does that do something werid? (Oh noo so many changed signals!). Alternatively I don't mind if we move this stuff over to BlockExtension itself. For BlockExtension I didn't want to add anything more than that one context_node property until it was needed, but as long as individual extensions aren't poking at BlockEditorContext it's fine.

Reply:
That was my initial instinct, but that's not what we want. If the BlockExtension watched for context changes, then the options would change every time you changed the block code node. We only want to do this in the picker where the blocks are reused for all contexts. I presume that's why you passed in the context node to BlockExtension instead of having it just grab the current context.

Reply:
Oh, yeah, I see where you're coming from there. Thanks for the explanation!